File: medshare/src/main/java/com/example/medshare/service/PatientService.java
package com.example.medshare.service;

import com.example.medshare.model.Patient;
import java.util.List;

public interface PatientService {
    Patient savePatient(Patient patient);
    Patient savePatient(Patient patient, String doctorId);
    List<Patient> getAllPatients();
    // 更多服务方法
    List<Patient> findPatientsByDoctorDoctorId(String doctorId);
    void deletePatient(String patientId, String doctorId);
    Patient updatePatient(Patient patient, String doctorId);

}

File: frontend/src/services/patientService.js
// src/services/patientService.js
import API from './api';

const addPatient = (patientData, doctorId) => {
  // 将doctorId添加到请求体中，以便后端能够识别
  return API.post(`/patients?doctorId=${doctorId}`, patientData);
};

const updatePatient = (patientData, doctorId) => {
  // 注意这里不再需要将patientId嵌入到URL中
  console.log("Updating patient with doctorId:", doctorId);
  return API.put(`/patients?doctorId=${doctorId}`, patientData);
};

const getAllPatients = () => {
  return API.get('/patients');
};

const getPatientsByDoctorId = (doctorId) => {
  return API.get(`/patients/by-doctor/${doctorId}`);
};

const deletePatient = (patientId, doctorId) => {
  return API.delete(`/patients/${patientId}?doctorId=${doctorId}`);
};


export { addPatient, updatePatient, getAllPatients, getPatientsByDoctorId, deletePatient };
